File: apps/android/feature/canvas/domain/src/main/java/com/scribblefit/feature/canvas/domain/model/WorkoutSession.kt
package com.scribblefit.feature.canvas.domain.model

data class WorkoutSession(
    val id: String,
    val startTime: Long,
    val lastActivityTime: Long,
    val exercises: List<SessionExercise>
)

data class SessionExercise(
    val canonicalName: String,
    val sets: List<SessionSet>
)

data class SessionSet(
    val weight: Double,
    val reps: Int,
    val rpe: Double? = null,
    val notes: String? = null
)

File: apps/ios/ScribbleFit/Sources/Features/Canvas/Domain/Models/WorkoutSession.swift
import Foundation

public struct WorkoutSession: Codable, Sendable {
    public let id: String
    public let startTime: Date
    public let lastActivityTime: Date
    public let exercises: [SessionExercise]
}

public struct SessionExercise: Codable, Sendable {
    public let canonicalName: String
    public let sets: [SessionSet]
}

public struct SessionSet: Codable, Sendable {
    public let weight: Double
    public let reps: Int
    public let rpe: Double?
    public let notes: String?
}
